### ISSUE TITLE
Upon login, update `context[:current_resource]` with `authenticatable`

### ISSUE THREAD

**Issue by TomasBarry** [graphql-devise/graphql_devise#192]
### What is the problem the enhancement will solve?

Given a schema, where I have changed `authenticable` to be the same type as `current_resource`, I would like to be able to query data on authenticable instead of having to send two queries - one for the mutation to log in to get the credentials and a second to perform the query to request the data that I want.

### Describe the solution you have in mind

In the base `Login` mutation, on [this](https://github.com/graphql-devise/graphql_devise/blob/master/lib/graphql_devise/mutations/login.rb#L26) line, add `context[:current_resource] = resource`.

### Describe alternatives you've considered

Sending two requests to the server - one for the mutation to log in and a second to query the data that I want.

Updating `context` prevents the need for two round trips.

### Additional context

If this is something you would like, I can open up a PR with the code changes/tests/documentation.

**Comment by mcelicalderon** [comment]
Hey @TomasBarry! I'm not sure I completely understand the problem. So you want to query information from `authenticatable` on the same login mutation, right? For that you should be all set, take a look at this example https://github.com/graphql-devise/graphql_devise/blob/a69b27a402933861304ebb35707b130092d15b4f/spec/requests/mutations/login_spec.rb#L20

Or not sure if you are trying to perform additional steps on the resource while the login mutation is executed. For that take a look at this example https://github.com/graphql-devise/graphql_devise/blob/a69b27a402933861304ebb35707b130092d15b4f/spec/dummy/app/graphql/mutations/login.rb#L8

Not sure I understand why you'd need to set current resource in the context of the login mutation, but again, I'm not completely sure I understand the problem. Perhaps you can share a sample query. Thanks!

**Comment by TomasBarry** [comment]
@mcelicalderon - I did leave out a crucial bit of information - If you have an schema using [Object Authorization](https://graphql-ruby.org/authorization/authorization.html#object-authorization) then this becomes a problem (by default, you're right it's not an issue).

Often, the object authorization uses the context:

```ruby
class Types::User < Types::BaseObject
  def self.authorized?(object, context)
    super && object == context[:current_resource]
  end
end
```

If you're using object authorization then `context[:current_resource]` is actually `nil`.

A way around this is to overload the `login` mutation.

Is this problem too use-case specific to be dealt with by the gem? If so I'm happy to close, I wanted to open this up before opening a PR that would be rejected.

**Comment by mcelicalderon** [comment]
@TomasBarry that makes sense! Happy to review your MR when you open it, thanks!

**Comment by mcelicalderon** [comment]
Released `v0.17.1` with your change, @TomasBarry. Thanks!
